archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -1575,16 +1576,17 @@ private Superclass(JavaType type) {
         }
 
         private Superclass(Optional<JavaType> type) {
-            this.rawType = type.map(TO_ERASURE);
+            this.rawType = type.map(TO_ERASURE::apply);
             this.type = type;
         }
 
         Optional<JavaClass> getRaw() {
             return rawType;
         }
 
+        @SuppressWarnings({"unchecked", "rawtypes"})
         Optional<JavaType> get() {
-            return type.or(rawType);
+            return type.isPresent() ? type : (Optional) rawType;
```

Comment:
cant we do `type.orElse(rawType)` here?

Reply:
The problem is, that we want to combine 2 `Optionals` and that doesn't seem to be possible in a convenient way :unamused: 

Starting with Java 9 there would be `optional.or(() -> otherOptional)`, but with Java 8 we would AFAIS have to do something ugly like

```
Optional.ofNullable(type.orElse(rawType.orElse(null)))
```

Don't have a strong preference, to me both looks bad :joy: Maybe I'll use this ugly piece of code though, just so we can get rid of the `@SuppressWarnings`, because that also is something we should really avoid :+1:

Reply:
I think "just so we can get rid of the @SuppressWarnings, because that also is something we should really avoid" is a totally reasonable way of thinking :D